articles/contributing/api-design-principles.md

```diff
@@ -0,0 +1,426 @@
+
+    - State preparation APIs published by Microsoft as a part of
+        the Quantum Development Kit would be placed into
+        `Microsoft.Quantum.Preparation`, while quantum simulation
+        APIs published by Microsoft as a part of the Quantum
+        Development Kit would be placed into
+        `Microsoft.Quantum.Simulation`.
+
+  - **DO** place operations, functions, and user-defined types used
+      only within specific domains into namespaces indicating their
+      domain of utility. If needed, use subnamespaces to indicate
+      focused tasks within each domain-specific namespace. For
+      example:
+
+    - Quantum chemistry APIs published by Microsoft as a part of
+        the Quantum Development Kit should be placed into
+        `Microsoft.Quantum.Chemistry`. Functionality specific to
+        implementing the Jordan--Wigner decomposition may be placed
+        in `Microsoft.Quantum.Chemistry.JordanWigner`, so that the
+        primary interface for the quantum chemistry domain area is
+        not concerned with implementations.
+
+- **DON'T** expose internal details related to implementation and
+    testing along with your API.
+
+  - Whenever reasonable, **DO** place all functions and operations
+      needed to implement an API into the same namespace as the API
+      being implemented, but marked with the "private" or "internal"
+      keywords to indicate that they are not part of the public API
+      surface for a library. Use a name beginning with an underscore
+      (`_`) to visually distinguish private and internal operations and
+      functions from public callables. For example:
+
+    - The operation name `_Features` indicates a function that is
+        private to a given namespace and assembly, and should be
+        accompanied by either the `private` or `internal` keyword.
+
+  - In the rare case that an extensive set of private functions or
+    operations are needed to implement the API for a given
+    namespace, **DO** place them in a new namespace matching the
+    namespace being implemented and ending in `.Private`.
+
+  - **DO** place all unit tests into namespaces matching the
+      namespace under test and ending in `.Tests`.
+
+## Naming Conventions and Vocabulary
+
+- **DO** choose names and terminology that are clear, accessible, and
+    readable across a diverse range of audiences, including both quantum
+    novices and experts.
+
+  - **DON'T** use discriminatory or exclusionary identifier names,
+      nor terminology in API documentation comments.
+
+  - **DO** use API documentation comments to provide relevant
+      context, examples, and references, especially for more difficult
+      concepts.
+
+  - **DON'T** use identifier names that are unnecessarily esoteric,
+      or that require significant quantum algorithms knowledge to
+      read. For example:
+
+    - Prefer a "amplitude amplification iteration" to "Grover
+        iteration."
```

Comment:
again too generic, and the example doesn't really help. for someone without quantum knowledge both terms sound equally esoteric.